app/scripts/modules/core/src/securityGroup/securityGroupReader.service.ts

```diff
@@ -216,6 +216,7 @@ export class SecurityGroupReader {
     const nameParts: IComponentName = NameUtils.parseSecurityGroupName(securityGroup.name);
     securityGroup.stack = nameParts.stack;
     securityGroup.detail = nameParts.freeFormDetails;
+    securityGroup.moniker = NameUtils.getMoniker(nameParts.application, nameParts.stack, nameParts.freeFormDetails);
```

Comment:
a fun bonus: now `ISecurityGroup` objects will always have a `moniker` on them, which is pretty nifty. they already have `stack` and `detail` fields that are empty strings when not present, which is kind of annoying and doesn't really look like anything else. I like the `moniker` object and it's contract around empty values being omitted from the object entirely.